File: C/bogosort/gemma2-9b-it.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

#define ARRAY_SIZE 1000

int main() {
  int arr[ARRAY_SIZE];
  srand(time(NULL));

  // Generate random input
  for (int i = 0; i < ARRAY_SIZE; i++) {
    arr[i] = rand() % 1000;
  }

  // Bogo Sort implementation
  int swapped;
  do {
    swapped = 0;
    for (int i = 0; i < ARRAY_SIZE - 1; i++) {
      for (int j = i + 1; j < ARRAY_SIZE; j++) {
        if (arr[i] > arr[j]) {
          int temp = arr[i];
          arr[i] = arr[j];
          arr[j] = temp;
          swapped = 1;
        }
      }
    }
  } while (swapped);

  return 0;
}

File: Java/bubblesort/gemma2_9b_it.java

public class gemma2_9b_it {

    public static void bubbleSort(int[] arr) {
        int n = arr.length;
        for (int i = 0; i < n - 1; i++) {
            for (int j = 0; j < n - i - 1; j++) {
                if (arr[j] > arr[j + 1]) {
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                }
            }
        }
    }

    public static void main(String[] args) {
        int[] arr = new int[1000];
        // Generate 1000 random numbers
        for (int i = 0; i < 1000; i++) {
            arr[i] = (int) (Math.random() * 1000);
        }
        bubbleSort(arr);
    }
}
